backend/src/server/routes/v1/user-router.ts

```diff
@@ -91,6 +92,75 @@ export const registerUserRouter = async (server: FastifyZodProvider) => {
     }
   });
 
+  server.route({
+    method: "GET",
+    url: "/list",
+    config: {
+      rateLimit: readLimit
+    },
+    schema: {
+      response: {
+        200: z.object({
+          users: UsersSchema.pick({
+            username: true,
+            firstName: true,
+            lastName: true,
+            email: true,
+            id: true
+          }).array()
+        })
+      }
+    },
+    onRequest: (req, res, done) => {
+      verifyAuth([AuthMode.JWT])(req, res, () => {
+        verifySuperAdmin(req, res, done);
+      });
+    },
+    handler: async () => {
+      const users = await server.services.user.listUsers();
+
+      return {
+        users
+      };
+    }
+  });
```

Comment:
I dont think any pagination is needed for this one because the intended usage is for dedicated/self-hosted instances

Reply:
Please add pagination as loading entire user list is a bad idea. Even though indention is self hosted if this page is opened in production this could turn up bad.

Another thing is please move this to admin route. All admin panel routing should go through admin service. This is to ensure as admin operations are peak security and should be kept different